src/deprecated.jl

```diff
@@ -213,3 +213,7 @@ export HDF5Attribute, HDF5File, HDF5Group, HDF5Dataset, HDF5Datatype,
 ### Changed in PR#691
 # - Make Reference() construct a null reference
 @deprecate_binding HDF5ReferenceObj_NULL HDF5.Reference()
+
+### Changed in PR#693
+const ScalarOrString = Union{ScalarType,String}
+@deprecate_binding ScalarOrString Union{ScalarType,String}
```

Comment:
```suggestion
@deprecate_binding ScalarOrString Union{ScalarType,String}
@deprecate_binding HDF5Scalar ScalarType
@deprecate_binding HDF5BitsKind BitsType
```

Reply:
Oh, also, the `const` line above isn't necessary, is it? The macro should generating the binding on demand.

Reply:
you're right, it's not necessary.